frontend/model/state.js

```diff
@@ -140,11 +144,45 @@ const mutations = {
   setChatRoomScrollPosition (state, { chatRoomId, messageId }) {
     Vue.set(state.chatRoomScrollPosition, chatRoomId, messageId)
   },
-  setChatRoomUnreadPosition (state, { chatRoomId, messageId, createdDate }) {
-    Vue.set(state.chatRoomUnreadPosition, chatRoomId, {
-      messageId, createdDate
+  deleteChatRoomScrollPosition (state, { chatRoomId }) {
+    Vue.delete(state.chatRoomScrollPosition, chatRoomId)
+  },
+  setChatRoomUnreadSince (state, { chatRoomId, messageId, createdDate }) {
+    const prevMentionings = state.chatRoomUnread[chatRoomId] ? state.chatRoomUnread[chatRoomId].mentionings : []
+    Vue.set(state.chatRoomUnread, chatRoomId, {
+      since: { messageId, createdDate, deletedDate: null },
+      mentionings: prevMentionings.filter(m => m.createdDate > createdDate) // TODO: in model/contracts/chatroom.js
+    })
+  },
+  deleteChatRoomUnreadSince (state, { chatRoomId, deletedDate }) {
+    Vue.set(state.chatRoomUnread[chatRoomId], 'since', {
+      ...state.chatRoomUnread[chatRoomId].since,
+      deletedDate
     })
   },
+  addChatRoomUnreadMentioning (state, { chatRoomId, messageId, createdDate }) {
+    const prevUnread = state.chatRoomUnread[chatRoomId]
+    if (!prevUnread) {
+      return
+    }
+    Vue.set(state.chatRoomUnread, chatRoomId, {
+      ...prevUnread,
+      mentionings: [...prevUnread.mentionings, { messageId, createdDate }]
+    })
```

Comment:
Does this really need to be written this way?

Won't work just the same if you do it like this?

```js
prevUnread.mentionings.push({ messageId, createdDate })
```